utils/codec/codec.go

```diff
@@ -316,7 +329,19 @@ func (c *codec) unmarshal(p *wrappers.Packer, value reflect.Value) error {
 		}
 		return nil
 	case reflect.Array:
-		for i := 0; i < value.Len(); i++ {
+		numElts := value.Len()
+		if elemKind := value.Type().Elem().Kind(); elemKind == reflect.Uint8 {
+			unpackedBytes := p.UnpackFixedBytes(numElts)
+			if p.Errored() {
+				return p.Err
+			}
+			for i := 0; i < numElts; i++ {
+				valAtIndex := value.Index(i)
+				valAtIndex.SetUint(uint64(unpackedBytes[i]))
+			}
```

Comment:
Is this as good as we can do here?

Reply:
Good point, changed this to take a slice of the underlying array and then copy the contents into that